### PR TITLE
Tiny fix

### PR DESCRIPTION
**Pull request by Skeletrox**
Seems like the delete permission code is working perfectly without this branch.

Create a user with no delete permissions and you should see the delete button vanish
The <i>actual</i> fix here was that a string was being split into individual chars that'd mess up the file upload process, so made it a list no matter how many extensions are being accepted

### CODE REVIEW

**Comment by rishabh1403** [comment]
Seems to work if I put .ecar,.json in input fields.
But initially the input fields show *Currently:[u'.ecar',u'.json']* , which is very misleading. It should show *Curently: .ecar,.json* so that user will know he has to provide input in that format

File: devmgmt/file_upload/changevars/views.py [hunk 1]
```diff
@@ -41,7 +41,10 @@ def update_data(request):
             new_val = request.POST.get("textinput-" + key, None)
             print 'Should be %s' %(new_val)
```
Comment (rishabh1403) [inline devmgmt/file_upload/changevars/views.py]: Comment out or remove the prints used while testing.

File: devmgmt/file_upload/changevars/views.py [hunk 2]
```diff
@@ -41,7 +41,10 @@ def update_data(request):
...
             if (new_val is not None and (len(new_val) > 0)):
-                varlist_full[key] = new_val
+				if key == 'accepted_extensions':
+					new_val = [t.strip() for t in new_val.split(',')]
+					print 'Bonjour %s' %(new_val)
```
Comment (rishabh1403) [inline devmgmt/file_upload/changevars/views.py]: Remove or comment unless needed in production

**Comment by Skeletrox** [comment]
Should be fixed now.

**Comment by chetansk** [comment]
Hi Sriram,

I hope you are doing a UT (unit test) before you send out the code for review. This is to make sure, reviewer's efforts are not wasted..

Thanks
C

**Comment by rishabh1403** [comment]
@Skeletrox  Not working as expected. Steps to reproduce.

- Login via an account where delete permission is not granted
- Either you will see the  delete button or won't. in that case upload a file
- The page will auto refresh and show delete button

**Comment by rishabh1403** [comment]
@Skeletrox Please use separate branches for separate issues. Delete button bug is not related to json bug in any way. The commits should have been in different branch. For example I cannot merge this branch in develop now just because it has changes that does not fix delete bug, even though it has changes  related to json bug which is fixed.

**Comment by Skeletrox** [comment]
@rishabh1403 I have no idea why it works for me and doesn't work for you. Even if I follow the steps the page refreshes and the delete button isn't there [Tested for both device upload as well as USB upload]. If the button shows up then there must be a delay in the transmission of permission data which causes this issue, as each permission is hidden only after permissions are transferred.

In retrospect, this is a security flaw and I've fixed it. Buttons only show up if the user is authorized to see them, isntead of showing them all and then hiding them. Tested for USB and file upload.

Apologies for any inconvenience

**Comment by rishabh1403** [comment]
@Skeletrox Attaching screenshots . Please note
- While logged in as root, change password is missing
- While logged in as other user, who does not have permissions to delete, delete button is visible
screenshot_20171204_165626
screenshot_20171204_165642

**Comment by Skeletrox** [comment]
Alright. The change password thing is an easy fix. However, the delete bug does not reciprocate in my machine. Here are my screencaps:

no delete permissions

no delete button

Please check your console to see if you're getting any errors.

**Comment by Skeletrox** [comment]
There was some issue in the upload tags that Django uses that I had fixed but had escaped a commit. Works fine now

**Pull request merged**